eventuate-cdc-data-producer-wrappers/src/main/java/io/eventuate/cdc/producer/wrappers/kafka/TopicPartitionSender.java

```diff
@@ -14,21 +15,21 @@
   private EventuateKafkaProducer eventuateKafkaProducer;
   private ConcurrentLinkedQueue<TopicPartitionMessage> messages = new ConcurrentLinkedQueue<>();
   private AtomicReference<TopicPartitionSenderState> state = new AtomicReference<>(TopicPartitionSenderState.IDLE);
-  private TopicPartitionMessage lastFailedMessage;
+  private boolean enableBatchProcessing;
+  private int batchSize;
 
-  public TopicPartitionSender(EventuateKafkaProducer eventuateKafkaProducer) {
+  public TopicPartitionSender(EventuateKafkaProducer eventuateKafkaProducer, boolean enableBatchProcessing, int batchSize) {
     this.eventuateKafkaProducer = eventuateKafkaProducer;
+    this.enableBatchProcessing = enableBatchProcessing;
+    this.batchSize = batchSize;
   }
 
   public CompletableFuture<?> sendMessage(String topic, String key, String body) {
```

Comment:
This code is a bit convoluted. message.poll() is called in multiple places.

I think sendMessage() and sendMessage(m) should be merged and look like

```
void sendMessage() {
if (!messages.isEmpty()) {
  sendNextMessage()
} else {
    state.set(TopicPartitionSenderState.IDLE);
    ...
}
```

sendNextMessage() should look like

```
if (!enableBatchProcessing) {
  sendSingleMessage()
else
  sendMessgeBatch()
```

and 

```
void sendMessgeBatch() {
    while (true) {
      TopicPartitionMessage messageForBatch = messages.peek();

      if (messageForBatch != null && messageBuilder
              .addMessage(new EventuateKafkaMultiMessageKeyValue(messageForBatch.getKey(), messageForBatch.getBody()))) {

        batch.add(messages.poll());
      }
      else {
        break;
      }
    if (batch.isEmpty()) {
        throw new RuntimeException("Sender is in error state, publishing is not possible.");
    }
 }
```

Reply:
ok